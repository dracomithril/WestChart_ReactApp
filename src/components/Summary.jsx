/**
 * Created by Gryzli on 09.04.2017.
 */
import React from "react";
import PropTypes from 'prop-types';
import {Badge} from "react-bootstrap";
let {sorting} = require('./../utils');
const create_print_list = (elem, index) => {
    return <div key={elem.id}>
        <span>{index + 1}</span>
        {`. ${elem.link.title} `}
        <Badge bsClass="likes">{elem.reactions_num + ' likes'}</Badge>
    </div>
};
export default class Summary extends React.Component {
    componentWillUnmount() {
        console.log('component Summary unmounted');
    }

    componentDidMount() {
        console.log('component Summary did mount');
    }

    render() {
        const {store} = this.context;
        const {list_sort} = store.getState();
        const sorting_options = Object.keys(sorting)
            .map((elem, index) => <option key={index} value={elem}>{elem.toLowerCase()}</option>);
        let print_list = this.props.selected.map(create_print_list);
        return (<div>
            <h3 id="list">{'List by: '}
                <select name="list_sort" value={list_sort}
                        onChange={(e) => store.dispatch({
                            type: 'UPDATE_LIST_SORT',
                            sort: e.target.value
                        })}>
                    {sorting_options}
                </select>
            </h3>
            <div id="popover-contained" title="Print list">
                {print_list}
            </div>
        </div>);
    }
}
Summary.contextTypes = {
    store: PropTypes.object
};
Summary.propTypes = {};