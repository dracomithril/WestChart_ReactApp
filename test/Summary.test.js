/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import Summary from '../src/components/Summary';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('./data/initial_state.json');

const mockStore = configureMockStore([]);


describe('<Summary/>', () => {
    it('renders without crashing', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <Summary selected={[]}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });
});
