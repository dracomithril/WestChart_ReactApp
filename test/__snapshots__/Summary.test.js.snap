// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary/> renders without crashing 1`] = `
<div>
  <h3
    id="list"
  >
    List by: 
    <select
      name="list_sort"
      onChange={[Function]}
      value="reaction"
    >
      <option
        value="reaction"
      >
        reaction
      </option>
      <option
        value="who"
      >
        who
      </option>
      <option
        value="when"
      >
        when
      </option>
      <option
        value="what"
      >
        what
      </option>
    </select>
  </h3>
  <div
    id="popover-contained"
    title="Print list"
  />
</div>
`;
